index.rb

```diff
@@ -11,7 +12,33 @@
   gem 'nokogiri'
 end
 
-Show = Struct.new(:time, :link, :title, :description, :venue, :keyword_init => true)
+Show = Struct.new(:time, :link, :title, :description, :venue, :keyword_init => true) do
+  def ical_link
+    'data:text/calendar;base64,' +
+    Base64.urlsafe_encode64(
+      [
+        ['BEGIN', 'VCALENDAR'],
+        ['VERSION', '2.0'],
+        ['BEGIN', 'VEVENT'],
+        ['URL', link],
+        ['DTSTART', ical_time(time)],
+        ['DTEND', ical_time(time + (3600 * 3))],
+        ['SUMMARY', title],
+        ['DESCRIPTION', description&.gsub("\n", '\\n')],
+        ['LOCATION', venue.name],
+        ['END', 'VEVENT'],
+        ['END', 'VCALENDAR']
+      ].map { |e| e.join(':') }.join("\n")
+    )
+  end
+
+  private
+
+  # https://icalendar.org/iCalendar-RFC-5545/3-3-5-date-time.html
+  def ical_time(time)
+    time.utc.strftime('%Y%m%dT%H%M%SZ')
```

Comment:
`utc` messed with our time objects, which is surprising: https://github.com/davishmcclurg/shows/commit/e61048b4493e58b4da4361ef211c47b0e9563380